plugins/undo/plugin.js

```diff
@@ -1080,6 +1087,9 @@
 			editable.attachListener( editable, 'paste', that.ignoreInputEventListener, that, null, 999 );
 			editable.attachListener( editable, 'drop', that.ignoreInputEventListener, that, null, 999 );
 
+			// After paste we need to re-enable input event listener
+			editor.on( 'afterPaste', that.activateInputEventListener, that );
```

Comment:
I wonder if there shouldn't be same priority for this listener as for ones above.